### PR TITLE
backend/#1516/cache bpp ride start and end geohash point

### PR DESCRIPTION
**Pull request by ashishsingh101**
## Type of Change

- [x] Bugfix
- [ ] New feature
- [ ] Enhancement
- [x] Refactoring
- [ ] Dependency updates

## Description

1. sending bpp ride start and end location to cache geohash in bap for hotspot 

### Additional Changes

- [ ] This PR modifies the database schema (database migration added)
- [ ] This PR modifies dhall configs/environment variables
- [ ] This PR contains API breaking changes

## Checklist

- [ ] I formatted the code and addressed linter errors `./dev/format-all-files.sh`
- [ ] I reviewed submitted code
- [ ] I added unit tests for my changes where possible
- [ ] I added a [CHANGELOG](/CHANGELOG.md) entry if applicable
- [ ] No leak detected in leakcanary

### CODE REVIEW

File: Backend/app/rider-platform/rider-app/Main/src/Beckn/ACL/OnUpdate.hs [hunk 1]
```diff
@@ -92,13 +93,33 @@ parseEvent _ (OnUpdate.RideAssigned taEvent) = do
...
       }
 parseEvent _ (OnUpdate.RideCompleted rcEvent) = do
   tagsGroup <- fromMaybeM (InvalidRequest "agent tags is not present in RideCompleted Event.") rcEvent.fulfillment.tags
+  let tripEndLat :: Maybe Double =
+        readMaybe . T.unpack
+          =<< getTag "ride_end_location" "ended_from_lat" tagsGroup
+  let tripEndLon :: Maybe Double =
+        readMaybe . T.unpack
+          =<< getTag "ride_end_location" "ended_from_lon" tagsGroup
+  let tripEndLocation = case (tripEndLat, tripEndLon) of
```
Comment (roman-bodavskiy) [inline Backend/app/rider-platform/rider-app/Main/src/Beckn/ACL/OnUpdate.hs:120]: can be simplified
`let tripEndLocation = Maps.LatLong <S> tripEndLat <*> tripEndLon`

Comment (ashishsingh101) [reply Backend/app/rider-platform/rider-app/Main/src/Beckn/ACL/OnUpdate.hs]: done

Comment (ashishsingh101) [reply Backend/app/rider-platform/rider-app/Main/src/Beckn/ACL/OnUpdate.hs]: done

Comment (ashishsingh101) [reply Backend/app/rider-platform/rider-app/Main/src/Beckn/ACL/OnUpdate.hs]: done

File: Backend/app/rider-platform/rider-app/Main/src/Beckn/ACL/OnUpdate.hs [hunk 2]
```diff
@@ -92,13 +93,30 @@ parseEvent _ (OnUpdate.RideAssigned taEvent) = do
         vehicleModel = vehicle.model
       }
 parseEvent _ (OnUpdate.RideStarted rsEvent) = do
+  personTagsGroup <- fromMaybeM (InvalidRequest "person tags is not present in RideStarted Event.") rsEvent.fulfillment.person.tags
```
Comment (RitikaHanish) [inline Backend/app/rider-platform/rider-app/Main/src/Beckn/ACL/OnUpdate.hs:96]: Don't throw error if it's not coming. Handle it well.

File: Backend/app/rider-platform/rider-app/Main/src/Beckn/ACL/OnUpdate.hs [hunk 3]
```diff
@@ -92,13 +93,30 @@ parseEvent _ (OnUpdate.RideAssigned taEvent) = do
...
   return $
     DOnUpdate.RideStartedReq
       { bppBookingId = Id rsEvent.id,
-        bppRideId = Id rsEvent.fulfillment.id
+        bppRideId = Id rsEvent.fulfillment.id,
+        ..
       }
 parseEvent _ (OnUpdate.RideCompleted rcEvent) = do
   tagsGroup <- fromMaybeM (InvalidRequest "agent tags is not present in RideCompleted Event.") rcEvent.fulfillment.tags
+  personTagsGroup <- fromMaybeM (InvalidRequest "person tags is not present in RideCompleted Event.") rcEvent.fulfillment.person.tags
```
Comment (RitikaHanish) [inline Backend/app/rider-platform/rider-app/Main/src/Beckn/ACL/OnUpdate.hs:112]: Don't throw error if it's not coming. Handle it well.

**Review by RitikaHanish — CHANGES_REQUESTED** [review]

**Review by RitikaHanish — APPROVED** [review]

**Pull request merged**